ccm_web/client/src/pages/Home.tsx

```diff
@@ -67,8 +68,18 @@ function Home (): JSX.Element {
 
   const cards: FeatureCardProps[] = [mergeSectionProps, gradebookToolsProps, createSectionsProps, addUMUsersProps, addNonUMUsersProps]
 
+  const saveCourseName = async (courseName: string): Promise<void> => await new Promise<void>((resolve, reject) => {
+    console.log('saveCourseName ' + courseName)
+    if (courseName === 'reject') {
+      reject(new Error('Rejected as requested'))
+    } else {
+      return resolve()
+    }
+  })
```

Comment:
The LTI needs to provide the Course Name after the LTI launch. So far it looks like the this will hold in React Props, once the course rename is happened then it need to persisted into props. Probably this can use react state lifting.

Reply:
I feel the saveCourseNamme function could be taken out Home, Probably into `InlineTextEdit`? Since all the req/res for features should be handled inside feature function. This is just a though No need to rework here. Since you are building UI with out API endpoint

Reply:
In my limited experience, I have found it more intuitive to have API interactions at the highest level in a container (here `Home`), and then passed down as needed to child components (somewhat related to the lifting state up idea). We can discuss, but I think having a consistent pattern for where API calls are set up and handled will make it easier to expand and modify.

Reply:
> n my limited experience, I have found it more intuitive to have API interactions at the highest level in a container (here Home), 

OK, valid this is a new territory and we will grow as we learn

But I feel since this is Home file, the API can exits here but the internals can be pulled out to a separate function, that way Home is not cluttered.

Reply:
Sure, I agree we shouldn't be `fetch`ing here, we can modularize it so it's relatively terse in this component.

Reply:
This was a placeholder for the sort of promise based functionality I was guessing would eventually be implemented, so I could build out the flow.  I don't think its super important to worry about where this code that is not going to be permanent lives at the moment.

Reply:
Sure, this is general discussion so that next time when actual implementing then you know team preference how this should be structured